### ISSUE TITLE
Load model from HF hub with a custom modeling file

### ISSUE THREAD

**Issue by ccdv-ai** [UKPLab/sentence-transformers#1473]
Hi,

I want to train a sentence embedding model that can handle long sequences.
For this I need to load a model from the hub that uses a custom modeling file.
However to load this [model](https://huggingface.co/ccdv/lsg-base-4096), I have to pass `trust_remote_code=True` to initialize the model with the AutoClass: 

```python
from sentence_transformers import SentenceTransformer, models

word_embedding_model = models.Transformer('ccdv/lsg-base-4096', max_seq_length=2048)

# fail because trust_remote_code=True isn't provided
```
The model should be loaded this way:
```python
model = AutoModel.from_pretrained('ccdv/lsg-base-4096', trust_remote_code=True)
```

Is adding additional initialization params planned ?

**Comment by toniopelo** [comment]
This seems to be implemented in a later release, I think this issue could be closed.
See [this PR](https://github.com/UKPLab/sentence-transformers/pull/2398), that was [released in v2.3.0](https://github.com/UKPLab/sentence-transformers/releases/tag/v2.3.0)
You might want to test it before closing the issue though, I didn't

**Comment by tomaarsen** [comment]
@toniopelo you're very right!  `trust_remote_code` can now be used with Sentence Transformers. I think this can indeed be closed - thanks for the help.

To load a Sentence Transformer model based on `ccdv/lsg-base-4096`:
```python
from sentence_transformers import SentenceTransformer

model = SentenceTransformer("ccdv/lsg-base-4096", trust_remote_code=True)
model.max_seq_length = 2048
```
And to load only a Transformer instance:
```python
from sentence_transformers import SentenceTransformer, models

word_embedding_model = models.Transformer('ccdv/lsg-base-4096', max_seq_length=2048, model_args={'trust_remote_code': True})
# e.g.:
pooling = models.Pooling(word_embedding_model.get_word_embedding_dimension())
model = SentenceTransformer(modules=[word_embedding_model, pooling])
```

I've tested this: it seems to work well.

- Tom Aarsen
